test/upload-result.test.js

```diff
@@ -0,0 +1,37 @@
+/*
+Copyright 2020 Adobe. All rights reserved.
+This file is licensed to you under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License. You may obtain a copy
+of the License at http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software distributed under
+the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+OF ANY KIND, either express or implied. See the License for the specific language
+governing permissions and limitations under the License.
+*/
+
+const should = require('should');
+
+const { importFile } = require('./testutils');
+
+const UploadResult = importFile('upload-result');
+const { setTimeoutPromise } = importFile('utils');
+
+describe('UploadResult Tests', function() {
+    it('test timer', async function() {
+        const uploadResult = new UploadResult();
+        uploadResult.startTimer();
+        await setTimeoutPromise(20);
```

Comment:
FYI, you can control the clock in tests using `sinon clock`, so you don't have to use sleeps/timeouts.
See documentation for details: https://sinonjs.org/releases/v1.17.6/fake-timers/

There are some nui tests that use the sinon clock to control time and verify timers. I can dig them up if you need more examples.

Reply:
Good suggestion, thank you. I'll look into the documentation.